src/containers/app/Apps.js

```diff
@@ -66,57 +87,306 @@ const ErrorMessage = styled.div`
-    this.props.getAppsRequest();
-    this.props.getOwnedOrganizations();
+    this.props.actions.getAppsRequest();
+    this.props.actions.fetchOrganizationsRequest();
   }
 
   componentWillReceiveProps(nextProps) {
     if (!this.props.organizations.size && nextProps.organizations.size) {
-      this.setState({ org: nextProps.organizations.keySeq().get(0)});
+      this.setState({ org: nextProps.organizations.keySeq().get(0) });
     }
   }
 
-  renderApps = () => {
-    return this.props.apps.map((app) => {
-      return (
-        <div key={app.get('name')}>
-          <AppSectionContainer>
+  onCreateApp = () => {
+    this.props.actions.createAppReset();
+    this.setState({
+      isAppModalOpen: true
+    });
+  };
+
+  onDeleteApp = (app) => {
+    this.props.actions.deleteAppRequest(app.get('id'));
+  };
+
+  onDeleteAppTypeFromApp = (appId, appTypeId) => {
+    this.props.actions.deleteAppTypeFromAppRequest(appId, appTypeId);
+  }
+
+  onAddAppTypeToApp = () => {
+    this.setState({
+      isAddAppTypeToAppModalOpen: true
+    });
+  }
+
+  renderAddAppForm = () => {
+    const { isError } = this.state;
+    return (
+      <form onSubmit={() => {
+        if (!this.state.addAppTypeAppTypeId) {
+          // Alert! Prevent form submission
+          this.setState({
+            isError: true
+          });
+          return;
+        }
+        this.props.actions.addAppTypeToAppRequest(this.state.addAppTypeAppId, this.state.addAppTypeAppTypeId);
+        this.closeModal();
+      }}>
+        <FormGroup>
+          <ControlLabel>Enter an App Type Id</ControlLabel>
+          <FormControl type="text" onChange={(e) => {
+            this.setState({ addAppTypeAppTypeId: e.target.value });
+          }} />
+        </FormGroup>
+        <br />
+        <Button type="submit" bsStyle="primary">Submit</Button>
+        { isError ? (<div style={{ color: 'red' }} >
+          <br />
+          Please check your inputs
+        </div>) : null}
+      </form>
+    );
+  }
+
+  onCreateAppType = () => {
+    this.props.actions.createAppTypeReset();
+    this.setState({
+      isAppTypeModalOpen: true
+    });
+  };
+
+  closeModal = () => {
+    this.setState({
+      isAppModalOpen: false,
+      isAppTypeModalOpen: false,
+      isAddAppTypeToAppModalOpen: false,
+      isEditAppModalOpen: false,
+      isEditAppTypeModalOpen: false
+    });
+    // FIGURE OUT HOW TO RESET THE MODAL STATE
+  };
+
+  getAppTypeIdByName = () => {
+    // REFERENCE ONLY
+    // for printing out an id for reference
+    const appTypeIdFromApi = AppApi.getAppTypeByFqn('sixth', 'apptype');
+    return appTypeIdFromApi;
+  }
+
+  renderAppType = (app) => {
+    // get the appTypeIds for the app
+    // for each appTypeId lookup the appTypes
+    const appTypeList = app.get('appTypeIds');
+    const appTypes = this.props.appTypes;
+    const appNames = [];
+    const { isEditAppTypeModalOpen } = this.state;
+
+    if (appTypes.get(appTypeList.get(0))) {
```

Comment:
What are you checking for here? First, `.get()` is not the right thing to have in a conditional; perhaps you want `.has()` Second, you're only checking `appTypeList.get(0)`, but below, you iterate through all elements in `appTypeList`.

If you want to check if `appTypes` is empty, you can just do `appTypes.isEmpty()`.